src/stores/store/secrets/hashicorp/hashicorp.go

```diff
@@ -180,19 +187,19 @@ func (s *Store) ListDeleted(_ context.Context) ([]string, error) {
 	return nil, err
 }
 
-func (s *Store) Restore(_ context.Context, id, version string) error {
-	if version == "" {
-		err := errors.NotSupportedError("secret version must be specified to restore")
-		s.logger.Warn(err.Error())
-		return err
+func (s *Store) Restore(ctx context.Context, id string) error {
+	logger := s.logger.With("id", id)
+	versionList, err := s.db.ListVersions(ctx, id, false)
```

Comment:
this block can be moved to a private function given that it logs so you can return the err directly and avoid duplicating the errMessage definition. Linting should fail anyways.

Reply:
You are right, I will encapsulate this as private method